src/utils/GitHubUtils/GitHubUtils.js

```diff
@@ -28,88 +28,45 @@ const getUserSvg = async (proxyServerUrl, gitHubUsername) => {
   };
 };
 
-export const setEmptyCalendarValues = calendar => {
-  const copiedCalendar = javaScriptUtils.deepCopyObject(calendar);
-
-  copiedCalendar.attributes = {
+export const initialize = produce(draft => {
+  draft.attributes = {
     viewBox: "0 0 828 128",
     preserveAspectRatio: "xMidYMin slice",
     class: "js-calendar-graph-svg",
     style:
       "width: 100%; padding-bottom: 14.5%; height: 1px; overflow: visible; font-size: .8em"
   };
 
-  copiedCalendar.children[0].children.forEach((weeklyData, weekIndex) => {
-    weeklyData.children.forEach((dailyData, dayIndex) => {
-      copiedCalendar.children[0].children[weekIndex].children[
-        dayIndex
-      ].attributes = {
-        ...dailyData.attributes,
-        "data-count": "0",
-        fill: "#ebedf0"
-      };
-    });
-  });
-
-  copiedCalendar.children[0].children.forEach(data => {
-    if (data.name === "text") {
-      // eslint-disable-next-line no-param-reassign
-      data.attributes = { ...data.attributes, fill: "#767676" };
+  draft.children[0].children.forEach((week, weekIndex) => {
+    if (week.name === "text") {
+      draft.children[0].children[weekIndex].fill = "#767676";
     }
-  });
 
-  return copiedCalendar;
-};
-
-export const mergeCalendarsContributions = (
-  actualCalendar,
-  gitHubUserJsonCalendar,
-  timeframe
-) => {
-  const copiedActualCalendar = javaScriptUtils.deepCopyObject(actualCalendar);
-
-  gitHubUserJsonCalendar.children[0].children.forEach(
-    (weeklyData, weekIndex) => {
-      weeklyData.children
-        .filter(dailyData =>
-          calendarUtils.filterContributionDays(dailyData, timeframe)
-        )
-        .forEach((dailyData, dayIndex) => {
-          const actualCalendarDailyData = calendarUtils.getCalendarDataByIndexes(
-            copiedActualCalendar,
-            weekIndex,
-            dayIndex
-          );
-          const totalDailyContributions =
-            Number(actualCalendarDailyData.attributes["data-count"]) +
-            Number(dailyData.attributes["data-count"]);
-
-          copiedActualCalendar.children[0].children[weekIndex].children[
-            dayIndex
-          ].attributes = {
-            ...actualCalendarDailyData.attributes,
-            "data-count": String(totalDailyContributions),
-            fill: calendarUtils.getFillColor(totalDailyContributions)
-          };
-        });
-    }
-  );
+    week.children.forEach((day, dayIndex) => {
+      draft.children[0].children[weekIndex].children[dayIndex].attributes[
+        "data-count"
+      ] = 0;
+      draft.children[0].children[weekIndex].children[dayIndex].attributes.fill =
+        "#ebedf0";
+    });
+  });
+});
 
-  return copiedActualCalendar;
-};
+export const dailyDataWithContributionsTransformation = calendar => {
+  // Extract the weeks from the calendar.
+  const weeks = calendar.children[0].children.slice(0, 53);
```

Comment:
One thing that'd make sense here is to extract the "magical" `53` number to a variable.

Reply:
Yes please, also shouldn't it be 52?

Reply:
Also same as above:

> This comment can very easily be replaced by a function. Always think if comments are really necessary.

The logic here looks very similar to `CalendarUtils`, couldn't that be used here?

Reply:
> Yes please, also shouldn't it be 52?

No, it's 53 (52 weeks + the current week).

> The logic here looks very similar to CalendarUtils, couldn't that be used here?

Hmmm, they're identical in that the both of them extract the first 53 elements from the calendar and iterates over these elements(two iterations, one for the weeks and one for the days). The difference lies in what they're doing with the weeks/days in each iteration.

Extracting the iterations to a function might make sense:

```js
const iterateCalendar = (calendar, cb) => {
   const weeks = extractWeeks(calendar);

   weeks.forEach(week => {
      week.children.forEach(day => {
         cb(day);
      });
   });
};

iterateCalendar(calendar, day => { ... });
```


But (IMO) it's just rather a "wrapper" and a "shorthand" version which replaces the double invoked `forEach()` calls.

Reply:
Fair enough